lib/termineter/modules/brute_force_login.py

```diff
@@ -57,61 +58,51 @@ def __iter__(self):
 				yield password
 				password = self.dictionary.readline()
 			self.dictionary.close()
-		raise StopIteration
+		#raise StopIteration; seems to have broken brute forcing multiple user id's
 
 def from_hex(data):
 	return binascii.a2b_hex(data)
 
 def to_hex(data):
 	return binascii.b2a_hex(data).decode('utf-8')
 
+
+
 class Module(TermineterModuleOptical):
 	connection_state = TermineterModuleOptical.connection_states.none
 	def __init__(self, *args, **kwargs):
 		TermineterModuleOptical.__init__(self, *args, **kwargs)
-		self.author = ['Spencer McIntyre']
+		self.author = ['Spencer McIntyre, David Garlak (@vexance)']
 		self.description = 'Brute Force Credentials'
 		self.detailed_description = 'This module is used for brute forcing credentials on the smart meter.  Passwords are not limited to ASCII values and in order to test the entire character space the user will have to provide a dictionary of hex strings and set USE_HEX to true.'
 		self.options.add_boolean('USE_HEX', 'values in word list are in hex', default=True)
 		self.options.add_rfile('DICTIONARY', 'dictionary of passwords to try', required=False, default='$DATA_PATH smeter_passwords.txt')
 		self.options.add_string('USERNAME', 'user name to attempt to log in as', default='0000')
 		self.options.add_integer('USER_ID', 'user id to attempt to log in as', default=1)
+		self.options.add_integer('USER_ID_MAX', 'maximum user id range to attempt bruteforce upon', default=50)
+		self.options.add_boolean('BRUTE_USERS', 'bruteforce user ids from USER_ID to USER_ID_MAX', default=False)
 
 		self.advanced_options.add_boolean('PURE_BRUTEFORCE', 'perform a pure bruteforce', default=False)
 		self.advanced_options.add_boolean('STOP_ON_SUCCESS', 'stop after the first successful login', default=True)
 		self.advanced_options.add_float('DELAY', 'time in seconds to wait between attempts', default=0.20)
 
-	def run(self):
+	def bruteforce_user(self, bf: BruteForce, current_id: int) -> bool:
+		self.frmwk.print_status(f'Attempting brute force on user id \'{current_id}\'')
```

Comment:
I'm pretty sure the f-strings were added in Python 3.6 and 3.4 is the oldest version that's still supported according to the setup.py file. 3.4 is ancient at this point so either the setup file should be updated, or this needs to be back-ported. I have no strong preference either way.

Reply:
Decided to back-port in case there is anything in other modules requiring 3.4